ibis/backends/datafusion/tests/test_register.py

```diff
@@ -25,24 +25,28 @@ def test_read_parquet(conn, data_dir):
 
 def test_register_table(conn):
     tab = pa.table({"x": [1, 2, 3]})
-    conn.register(tab, "my_table")
-    assert conn.table("my_table").x.sum().execute() == 6
+    with pytest.warns(FutureWarning, match="v9.0"):
+        conn.register(tab, "my_table")
+        assert conn.table("my_table").x.sum().execute() == 6
```

Comment:
```suggestion
    with pytest.warns(FutureWarning, match="v9.0"):
        conn.register(tab, "my_table")
    assert conn.table("my_table").x.sum().execute() == 6
```

We should only have the line (or lines) that warns in the context manager block